bindings/java/src/test/java/org/openucx/jucx/UcpMemoryTest.java

```diff
@@ -71,11 +74,49 @@ public void testRemoteKeyUnpack() {
         UcpMemory mem = context.registerMemory(buf);
         UcpRemoteKey rkey = endpoint.unpackRemoteKey(mem.getRemoteKeyBuffer());
         assertNotNull(rkey.getNativeId());
-        rkey.close();
-        mem.deregister();
-        endpoint.close();
-        worker1.close();
-        worker2.close();
-        context.close();
+
+        Collections.addAll(resources, context, worker1, worker2, endpoint, mem, rkey);
+        closeResources();
+    }
+
+    @Test
+    public void testGPUOperations() throws Exception {
+        UcpParams params = new UcpParams()
+            .requestTagFeature().requestAmFeature();
+        UcpContext context1 = new UcpContext(params);
+        if (!UcsConstants.MEMORY_TYPE.isMemTypeSupported(context1.getMemoryTypesMask(),
+             UcsConstants.MEMORY_TYPE.UCS_MEMORY_TYPE_CUDA)) {
```

Comment:
can we run same test on multiple memory types (host, cuda, cuda manager)? 
memory type could be a parameter to helper function

Reply:
Done: https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=13866&view=logs&jobId=2d17f13d-f402-5448-dd96-eeac985c8863&j=2d17f13d-f402-5448-dd96-eeac985c8863&t=61d25ca5-2380-522c-68e5-91463ebbd3bf

```
Running testIovOperations with memType: 0
Running testIovOperations with memType: 1
Running testIovOperations with memType: 2
Running testGetNB with memType: 0
Running testGetNB with memType: 1
Running testGetNB with memType: 2
Running testPutNB with memType: 0
Running testPutNB with memType: 1
Running testPutNB with memType: 2
Running testActiveMessages with memType: 0
Running testActiveMessages with memType: 1
Running testActiveMessages with memType: 2
Running testSendRecv with memType: 0
Running testSendRecv with memType: 1
Running testSendRecv with memType: 2
```

For now send/recv pair is the same